### ISSUE TITLE
Do you realized key_padding_mask in you long-term attention and short-term Attention?

### ISSUE THREAD

**Issue by zhouweii234** [z-x-yang/AOT#3]
Do you realized `key_padding_mask` (a parameter in `torch.nn.multiheadattention`) in the long-term attention and short-term Attention? Or you just do not need this in your network?

**Comment by z-x-yang** [comment]
No. The key_padding_mask is useless for long-term attention and short-term.

**Comment by zhouweii234** [comment]
Thank you for your reply! By the way , do you use `attn_mask`?

**Comment by z-x-yang** [comment]
Long-term attention doesn't need any mask. The implementation of short-term attention is complicated. Since short-term attention has some processes related to the relative positions between pixels. Masking out the correlations out of feature boundary is necessary.
